google-cloud-spanner-hibernate-testing/src/test/java/org/hibernate/jpa/test/BaseEntityManagerFunctionalTestCase.java

```diff
@@ -286,16 +270,43 @@ private void releaseUnclosedEntityManager(EntityManager em) {
 
 		if ( em.getTransaction().isActive() ) {
 			em.getTransaction().rollback();
-            log.warn("You left an open transaction! Fix your test case. For now, we are closing it for you.");
+			System.out.println(
+					"You left an open transaction! Fix your test case. For now, we are closing it for you.");
 		}
 		if ( em.isOpen() ) {
 			// as we open an EM before the test runs, it will still be open if the test uses a custom EM.
 			// or, the person may have forgotten to close. So, do not raise a "fail", but log the fact.
 			em.close();
-            log.warn("The EntityManager is not closed. Closing it.");
+			System.out.println("The EntityManager is not closed. Closing it.");
+		}
+	}
+
+	protected void doIfNotInitialized(Runnable runnable) {
+		if (!isInitialized) {
+			runnable.run();
+			isInitialized = true;
 		}
 	}
 
+	@Before
+	@SuppressWarnings({"UnusedDeclaration"})
+	public void buildEntityManagerFactory() {
```

Comment:
Can you explain a bit more what the changes are?

Reply:
Added explanation in code, PTAL